### PR TITLE
Multi Runtime

### PR DESCRIPTION
**Pull request by t0rr3sp3dr0**
Allow operator to properly run on a mixed runtime enviroment, such as a K8s cluster with nodes of different architectures.

### CODE REVIEW

File: operator/controllers/actionsrunner/util/converters.go [hunk 1]
```diff
@@ -360,22 +360,41 @@ func ToJob(actionsRunner *inlocov1alpha1.ActionsRunner, actionsRunnerJob *inloco
...
 	}
+	nodeAffinity := affinity.NodeAffinity
 
-	return resourceRequirements
+	if nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+		nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &corev1.NodeSelector{}
+	}
+	nodeSelector := nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution
+
+	nodeSelector.NodeSelectorTerms = append(nodeSelector.NodeSelectorTerms, corev1.NodeSelectorTerm{
```
Comment (rhnasc) [inline operator/controllers/actionsrunner/util/converters.go:378]:
```
If you specify multiple nodeSelectorTerms associated with nodeAffinity types, then the pod can be scheduled onto a node if one of the nodeSelectorTerms can be satisfied.
```
https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/

Is this the intended behavior?

File: operator/controllers/actionsrunner/util/filters.go [hunk 2]
```diff
@@ -45,3 +45,21 @@ func FilterEnv(oldEnv []corev1.EnvVar, predicate func(corev1.EnvVar) bool) []cor
...
 }
+
+func FilterResources(source corev1.ResourceRequirements, resourceNames ...corev1.ResourceName) corev1.ResourceRequirements {
+	resourceRequirements := corev1.ResourceRequirements{
+		Limits:   corev1.ResourceList{},
+		Requests: corev1.ResourceList{},
+	}
+
+	for _, resourceName := range resourceNames {
+		if val, ok := source.Limits[resourceName]; ok {
```
Comment (rhnasc) [inline operator/controllers/actionsrunner/util/filters.go:56]: Does source ever come null? You may want to add a guard clause at the beginning of method

File: operator/controllers/actionsrunner/util/converters.go [hunk 3]
```diff
@@ -360,22 +360,41 @@ func ToJob(actionsRunner *inlocov1alpha1.ActionsRunner, actionsRunnerJob *inloco
 	return &job, nil
 }
 
-func filterResourceRequirements(source corev1.ResourceRequirements, resourceNames ...corev1.ResourceName) corev1.ResourceRequirements {
-	resourceRequirements := corev1.ResourceRequirements{
-		Limits:   corev1.ResourceList{},
-		Requests: corev1.ResourceList{},
+func withRuntimeAffinity(affinity *corev1.Affinity) *corev1.Affinity {
+	if affinity == nil {
```
Comment (rhnasc) [inline operator/controllers/actionsrunner/util/converters.go:364]: This long list of `if`s hurts my eyes x.x Unfortunately this seems like the best we can do

File: operator/controllers/actionsrunner/util/converters.go (hunk 1 above)
Comment (t0rr3sp3dr0) [reply operator/controllers/actionsrunner/util/converters.go:378]: Not actually. I will change it to append/override `kubernetes.io/os` and `kubernetes.io/arch` on existent NodeSelectorTerms and if none exists I will create one with the requirements.

File: operator/controllers/actionsrunner/util/filters.go (hunk 2 above)
Comment (t0rr3sp3dr0) [reply operator/controllers/actionsrunner/util/filters.go:56]: No, it doesn't. `source` is a struct, not a pointer, then it is always valid.

**Pull request merged**